distributed/security.py

```diff
@@ -326,6 +326,8 @@ def _get_tls_context(self, tls, purpose):
             # IP addresses rather than hostnames
             ctx.check_hostname = False
 
+            ctx.verify_flags &= ~ssl.VERIFY_X509_STRICT
```

Comment:
I've gone ahead and turned this new `VERIFY_X509_STRICT` flag off here (it was previously off by default in older versions of Python but was flipped on by default in Python 3.13). This seems not ideal, but also not that bad given that's what we've always done prior to Python 3.13. @jacobtomlinson (or anyone else) let me know if you have a better suggestion, I'm not verify familiar with our TLS security code paths.

Reply:
using https://github.com/python-trio/trustme to generate the certificates used for testing on the fly would get this to pass (as it sets the appropriate flags on the certificates and generally keeps these flags up to date)

of course, distributed users would also need to configure their certificates to meet these stronger security standards